jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/HTTP2Connection.java

```diff
@@ -300,6 +309,12 @@ public void onConnectionFailure(int error, String reason)
         session.onConnectionFailure(error, reason);
     }
 
+    @Override
+    public String toConnectionString()
```

Comment:
This method is eventually called by `AbstractEndPoint.toString()` so you cannot take a lock in it, or more precisely in this case: you cannot call a method that takes a lock.